controllers/machineset_controller.go

```diff
@@ -230,7 +230,7 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 
 	var errs []error
 	for _, machine := range filteredMachines {
-		if conditions.IsFalse(machine, clusterv1.MachineOwnerRemediatedCondition) {
+		if machine.DeletionTimestamp.IsZero() && conditions.IsFalse(machine, clusterv1.MachineOwnerRemediatedCondition) {
```

Comment:
Instead of adding this to the `if` statements reserved to remediation, let's add `if !machine.DeletionTimestamp.IsZero() { continue }` above? 

Then we can also add a comment around it explaining why we're excluding the deleted ones